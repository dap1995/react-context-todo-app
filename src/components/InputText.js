// @flow
import React from 'react';
import styled from 'styled-components';

const Input = styled.input`
  flex: 1;
  padding: 0.5em;
  margin: 0.5em;
  border: none;
  border-radius: 3px;
  font-size: 24px;

  &::placeholder {
    font-style: italic;
    font-weight: 300;
    color: #e6e6e6;
  }

  &:focus {
    outline: 0;
    border:1px solid rgba(175,47,47,0.15);
    box-shadow: 0 0 10px #e6e6e6;
  }
`;

type Props = {
  onSubmit: (text: string) => void
};
type State = {
  text: string
};
type Event = {
  key: string,
  target: {
    value: string
  }
};

/**
 * Simple InputText component to submit on press `Enter` key.
 * Receives a bound method on props, called `onSubmit`.
 */
class InputText extends React.PureComponent<Props, State> {
  state = {
    text: ''
  };

  // Submit the input text if has pressed enter.
  handleKeyPress = (evt: Event) => {
    if (evt.key === 'Enter') {
      const { text } = this.state;
      this.props.onSubmit(text);
      this.setState({ text: '' });
    }
  };

  onChange = (evt: Event) => {
    this.setState({ text: evt.target.value });
  };

  render() {
    return (
      <Input
        type="text"
        placeholder="What needs to be done?"
        value={this.state.text}
        onKeyDown={this.handleKeyPress}
        onChange={this.onChange}
        autoFocus={true}
      />
    );
  }
}

export default InputText;
