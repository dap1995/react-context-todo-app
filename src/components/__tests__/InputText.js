import React from 'react';
import ReactDOM from 'react-dom';
import InputText from '../InputText';

it('renders without crashing', () => {
  const input = document.createElement('input');
  ReactDOM.render(<InputText />, input);
});
