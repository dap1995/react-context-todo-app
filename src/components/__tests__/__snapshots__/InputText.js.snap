// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be the same as the snapshot 1`] = `
<input
  autoFocus={true}
  className="sc-bdVaJa bhfcXg"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="What needs to be done?"
  type="text"
  value=""
/>
`;
